cloudify_aws/ec2/resources/ebs.py

```diff
@@ -204,58 +248,95 @@ def delete(ctx, iface, resource_config, **_):
+                                resource_id=utils.get_resource_id(
+                                    node=ctx.source.node,
+                                    instance=ctx.source.instance,
+                                    raise_on_missing=True))
+
+    _attach_ebs(params, iface, ctx.source)
+
+
+@decorators.aws_relationship(EC2Volume, RESOURCE_TYPE_VOLUME)
+@decorators.wait_on_relationship_status(status_good=[DETACHED, AVAILABLE],
+                                        status_pending=[DETACHING, INUSE])
+def detach_from(ctx, iface, **_):
+    """
+    De-attaches an AWS EC2 EBS Volume TO Instance
+    :param ctx:
+    :param iface:
+    :param _:
+    """
+    iface = EC2VolumeAttachment(ctx.source.node, logger=ctx.logger,
+                                resource_id=utils.get_resource_id(
+                                    node=ctx.source.node,
+                                    instance=ctx.source.instance,
+                                    raise_on_missing=True))
+    resource_id = \
+        ctx.source.instance.runtime_properties[constants.EXTERNAL_RESOURCE_ID]
+    _detach_ebs(iface, resource_id)
+
+
 @decorators.aws_resource(EC2VolumeAttachment, RESOURCE_TYPE_VOLUME_ATTACHMENT)
 @decorators.wait_for_status(status_good=[ATTACHED, INUSE],
                             status_pending=[ATTACHING])
-def attach(ctx, iface, resource_config, **_):
+def create_attachment(ctx, iface, resource_config, **_):
     """
     Attaches an AWS EC2 EBS Volume TO Instance
     :param ctx:
     :param iface:
     :param resource_config:
     :param _:
-    :return:
     """
-    params = \
-        dict() if not resource_config else resource_config.copy()
-
-    # Attach ebs volume to ec2 instance resource
-    create_response = iface.create(params)
-
-    # Check if the resource attaching done
-    if create_response:
-        ctx.instance.runtime_properties['eps_attach'] =\
-            utils.JsonCleanuper(create_response).to_dict()
-
-        # Update the esp_id (volume_id)
-        esp_id = create_response.get(VOLUME_ID, '')
-        utils.update_resource_id(ctx.instance, esp_id)
-        iface.update_resource_id(esp_id)
-
-    else:
-        raise NonRecoverableError(
-            '{0} ID# "{1}" reported an empty response'
-            .format(RESOURCE_TYPE_VOLUME_ATTACHMENT, iface.resource_id))
+    params = dict() if not resource_config else resource_config.copy()
+    response = _attach_ebs(params, iface, ctx)
+    # Update the esp_id (volume_id)
+    esp_id = response.get(VOLUME_ID, '')
+    utils.update_resource_id(ctx.instance, esp_id)
+    iface.update_resource_id(esp_id)
 
 
 @decorators.aws_resource(EC2VolumeAttachment, RESOURCE_TYPE_VOLUME_ATTACHMENT,
                          ignore_properties=True)
 @decorators.wait_for_status(status_good=[DETACHED, AVAILABLE],
                             status_pending=[DETACHING, INUSE])
-def detach(ctx, iface, resource_config, **_):
+def delete_attachment(ctx, iface, **_):
```

Comment:
We don't need to keep the old name, but we should have a function with the old name that calls the new function in case someone is using it. OK?

Reply:
Make sense